daemon/k8s_watcher.go

```diff
@@ -1383,8 +1316,35 @@ func updateCNPNodeStatus(ciliumNPClient clientset.Interface, cnp *cilium_v2.Cili
 			_, err2 = ciliumNPClient.CiliumV2().CiliumNetworkPolicies(ns).Patch(cnp.GetName(), types.JSONPatchType, createStatusAndNodePatchJSON, "status")
 		}
 	default:
-		cnp.SetPolicyStatus(nodeName, cnpns)
-		_, err2 = ciliumNPClient.CiliumV2().CiliumNetworkPolicies(ns).Update(cnp)
+		// During testing it was noticed kube-apiserver got panics when this
+		// patch was performed, for this reason we will not use node PatchJson.
+		_, err2 = ciliumNPClient.CiliumV2().CiliumNetworkPolicies(ns).Patch(cnp.GetName(), types.JSONPatchType, createStatusAndNodePatchJSON)
+		if err2 != nil {
+			// If it fails it might mean the `/status` path exists.
+			// Before you ask, we can't blindly add this node status to the path
+			// `/status` because it will delete all other nodes.
+			createStatusAndNodePatch[0].Value = cilium_v2.CiliumNetworkPolicyStatus{}
```

Comment:
Add a comment to indicate that this updates the `test` portion of the `createStatusAndNodePatch` to test if the NodeStatus is empty instead of `nil`.